CONTRIBUTING.md

```diff
@@ -47,6 +47,92 @@ Using commit signoffs tags is mandatory for all commits; we also
 encourage that each commit is small and cohesive. See the next sections for details.
 
 
+### Programming style
+
+#### Organization
+
+All code that interacts with the operating system, such as file access, network
+operations, timers, process creation, etc, must happen through a C++
+interface. These interfaces must be organized in a
+`common/<MODULE>/<MODULE>.hpp` hierarchy. The platform implementation must
+reside in `common/<MODULE>/impl/<MODULE>.cpp` files. It is allowed to use more
+files in addition to the `<MODULE>` named files, if necessary.
+
+All the API must be namespaced inside the `<MODULE>` name. Avoid C macros if
+possible, since they can't be namespaced.
+
+#### Code style
```

Comment:
I personally would prefer to have this as the first paragraph.

Reply:
Ok, I moved it.